File: Services/dotnet-services/Core/OmniCure.Core.Domain/Interfaces/Persistence/IQueryExecutor.cs
using OmniCure.Core.Shared.Pagination;

namespace OmniCure.Core.Domain.Interfaces.Persistence;

public interface IQueryExecutor
{
    Task<PagedList<TResult>> ToPagedListAsync<TResult>(
        IQueryable<TResult> query,
        int pageNumber,
        int pageSize,
        CancellationToken cancellationToken = default);
}

File: Services/java-services/Core/CoreDomain/src/main/java/ai/omnicure/core/domain/repository/IQueryExecutor.java
package ai.omnicure.core.domain.repository;

import ai.omnicure.core.shared.pagination.PagedList;
import org.springframework.data.jpa.repository.JpaRepository;

public interface IQueryExecutor {
    <T> PagedList<T> toPagedList(Iterable<T> items, long totalCount, int pageNumber, int pageSize);
}
